src/DraftEditor.tsx

```diff
@@ -534,7 +534,7 @@ class DraftEditor extends Component<IDraftEditorProps, IDraftEditorState> {
             return 'handled';
         }
         if (valueSearchOpen) {
-            if (e.key === 'Enter') {
+            if (e.key === 'Enter' && document.querySelector('.value-mention-item-focused')) {
```

Comment:
assign document.querySelector('.value-mention-item-focused') to a variable and use